app/forms/DateOfLeavingForm.scala

```diff
@@ -16,14 +16,23 @@
 
 package forms
 
+import com.google.inject.Singleton
 import models.{GmpDate, Leaving}
-import play.api.Play.current
+import play.api.Play
 import play.api.data.Form
 import play.api.data.Forms._
 import play.api.data.validation.{Constraint, Invalid, Valid, ValidationError}
-import play.api.i18n.Messages
-import play.api.i18n.Messages.Implicits._
-object DateOfLeavingForm {
+import play.api.i18n.{Messages, MessagesApi, MessagesImpl}
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.mvc.MessagesControllerComponents
+@Singleton
+class BaseDateOfLeavingForm  {
+
+  val messagesControllerComponents = Play.current.injector.instanceOf[MessagesControllerComponents]
```

Comment:
try class level injection instead of Play.current

Reply:
messagesApi will be available from mcc

Reply:
maybe we can leave it as it is and refactor as part of removing compiler warnings